src/js/component.js

```diff
@@ -1072,16 +1069,16 @@ Component.prototype.emitTapEvents = function(){
     } else if (firstTouch) {
       // Some devices will throw touchmoves for all but the slightest of taps.
       // So, if we moved only a small distance, this could still be a tap
-      xdiff = event.touches[0].pageX - firstTouch.pageX;
-      ydiff = event.touches[0].pageY - firstTouch.pageY;
-      touchDistance = Math.sqrt(xdiff * xdiff + ydiff * ydiff);
+      let xdiff = event.touches[0].pageX - firstTouch.pageX;
+      let ydiff = event.touches[0].pageY - firstTouch.pageY;
+      let touchDistance = Math.sqrt(xdiff * xdiff + ydiff * ydiff);
       if (touchDistance > tapMovementThreshold) {
         couldBeTap = false;
       }
     }
   });
 
-  noTap = function(){
+  let noTap = ()=> {
```

Comment:
I don't know how I missed these, but they're the same as the other example we reverted. I looked at how babel translates arrow functions and it at least doesn't do an extra `fn.bind()`, so that's good. I want to make sure I'm not just being a luddite here so if there's fault in my reasoning let me know. I think it's an important issue.

I still don't think the shortened syntax alone is benefit enough to warrant upping the javascript reading level for contributors. For most contributors JavaScript is probably still not their primary programming lang, and we'd be replacing one of the most most well known keywords with one of the least known that does the exact same thing (in this specific case).

Also, Babel makes named functions when translating arrow functions.
```
var noTap = function noTap() {}
```
That goes against the "Arrow functions are always anonymous" rule and could potentially [create issues in IE8](http://kiro.me/blog/nfe_dilemma.html). Though I'm not sure how likely that is and I wish we could use named functions more for stack traces.

Also if you have a text editor with a function shortcut, there's not much benefit left over for this specific case. `f` then `tab` quickly in Atom makes an empty function.

I do think there are some cool examples of using arrow functions, like:
```
let square = x => x * x;
```
I would have a harder time arguing against that, and it even seems more readable than the example above for some reason.

And using it for it's `this` binding.
```
var boundFn = ()=> {
  this.foo();
}
```

But in all the examples here we seem to just be using it for it's shorter anonymous function syntax, which if that's correct then for consistency there's over 500 other anonymous functions in the codebase we would need to update too. Am I missing another benefit in these examples?